fs/layer/layer.go

```diff
@@ -325,6 +330,8 @@ func (r *Resolver) Resolve(ctx context.Context, hosts source.RegistryHosts, refs
 	log.G(ctx).Debugf("[Resolver.Resolve]Initialized metadata store for layer sha=%v", desc.Digest)
 
 	spanManager := spanmanager.New(ztoc, sr, spanCache, cache.Direct())
+	bgLayerResolver := backgroundfetcher.NewSequentialResolver(desc.Digest, spanManager)
+	r.bgFetcher.Add(bgLayerResolver)
```

Comment:
There should probably be someway to unregister a `layerResolver` (e.g. because the container exited). We already call `layer.Done()` when the layer is unmounted so we should probably hook the background layer resolver into that.

Reply:
Makes sense. I don't think `Done()` is the right place to do so, because it's called whenever a ref count in the lrucache is decreased. This means that layer can still be mounted, but we won't do background fetch anymore. Example: https://github.com/awslabs/soci-snapshotter/blob/192f026b6267a08241cfe7e9e9f1346b81a49ccc/fs/fs.go#L384-L386

Instead, decided to close the resolver in `layer.close()`.